cinema/views.py

```diff
@@ -124,6 +161,17 @@
     )
     serializer_class = OrderSerializer
     pagination_class = OrderPagination
+    authentication_classes = (TokenAuthentication,)
+
+    def get_permissions(self):
+        if self.action in ("list", "create"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        elif self.action in ("retrieve", "update", "destroy"):
+            permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
+        else:
+            raise NotFound("Method not allowed for this resource")
```

Comment:
Raising `NotFound` here is incorrect. Use `MethodNotAllowed` to indicate that the method is not allowed for the requested resource.